modules/telemetry/telemetry.py

```diff
@@ -70,7 +65,7 @@ def __init__(
 
         # Telemetry Data holds the last few copies of received data blocks stored under the subtype name as a key.
         self.status: jsp.StatusData = jsp.StatusData()
-        self.telemetry: dict[str, list[dict[str, str]]] = {}
+        self.telemetry: jsp.TelemetryData = jsp.TelemetryData(self.config.telemetry_buffer_size)
```

Comment:
Can we make the naming of this more explicit? `self.telemetry` is a confusing name when the class name is also `telemetry`. Maybe something like `self.data_buffer`?

Reply:
self.telemetry_data 
👍